tests/components/climate/test_reproduce_state.py

```diff
@@ -117,3 +117,57 @@ async def test_attribute(hass, service, attribute):
 
     assert len(calls_1) == 1
     assert calls_1[0].data == {"entity_id": ENTITY_1, attribute: value}
+
+
+async def test_attribute_partial_temperature(hass):
+    """Test that service call ignores null attributes."""
+    calls_1 = async_mock_service(hass, DOMAIN, SERVICE_SET_TEMPERATURE)
+
+    await async_reproduce_states(
+        hass,
+        [
+            State(
+                ENTITY_1,
+                None,
+                {
+                    ATTR_TEMPERATURE: 23.1,
+                    ATTR_TARGET_TEMP_HIGH: None,
+                    ATTR_TARGET_TEMP_LOW: None,
```

Comment:
IMHO, we should filter the resulting service calls to only pass data values that match the entity capabilities.

Reply:
I need to understand in more detail.

I understand this to mean a check for `SUPPORT_TARGET_TEMPERATURE` (or more generally the corresponding support flag for each attribute). That makes sense to me.


I think this could mean it should be handled on the side where the state is captured.


When considering the fchexk at scene save time, even if it's supported (e.g. 
`SUPPORT_TARGET_TEMPERATURE` is present) then the `None` value here needs to be omitted since the service call is failing due to a `None` passed in where it expects a float.

I realize now in nest example, the supported attributes depend on other attributes like the HVAC mode. I wonder if that may means we need to refresh the current entity state as previous attributes are restored (e.g. after HVAC mode is set)

Reply:
My impression is the problem here is related to some templating even before making the actual service call. When I make this service call in the developer console:
```
service: climate.set_temperature
data:
  temperature: 78
  target_temp_high:
  target_temp_low:
target:
  entity_id: climate.dining_room
```
Then I get back this error which I assumed was before we're even getting to an entity or its capabilities:
```
2021-07-26 14:46:19 ERROR (MainThread) [homeassistant.helpers.script.websocket_api_script] websocket_api script: Error executing script. Invalid data for call_service at pos 1: expected float for dictionary value @ data['target_temp_high']
```

I also have the impression that state attributes are not necessarily related to its capabilities.  That is, a climate entity can support various target temperatures even if they aren't currently set at the moment. (e.g. capabilities are based on supported hvac modes, but target temp is independent).

I am likely missing a subtlety of your point and elaboration could help me understand. Thanks!

Reply:
You are right, the schema of the service call would not accept it, thus this solution you implement here is the right one.